pkg/test/e2e/encryption-at-rest/ear_test.go

```diff
@@ -0,0 +1,906 @@
+	"k8c.io/kubermatic/v2/pkg/log"
+	"k8c.io/kubermatic/v2/pkg/resources"
+	"k8c.io/kubermatic/v2/pkg/resources/encryption"
+	"k8c.io/kubermatic/v2/pkg/test/e2e/jig"
+	"k8c.io/kubermatic/v2/pkg/test/e2e/utils"
+	"k8c.io/kubermatic/v2/pkg/util/podexec"
+	"k8c.io/kubermatic/v2/pkg/util/wait"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlruntimelog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var (
+	credentials jig.BYOCredentials
+	logOptions  = utils.DefaultLogOptions
+)
+
+func init() {
+	credentials.AddFlags(flag.CommandLine)
+	jig.AddFlags(flag.CommandLine)
+	logOptions.AddFlags(flag.CommandLine)
+}
+
+const (
+	defaultTimeout         = 10 * time.Minute
+	defaultInterval        = 5 * time.Second
+	minioBackupDestination = "minio"
+)
+
+type runner struct {
+	seedClient ctrlruntimeclient.Client
+	userClient ctrlruntimeclient.Client
+	config     *rest.Config
+	logger     *zap.SugaredLogger
+	testJig    *jig.TestJig
+}
+
+func TestEncryptionAtRest(t *testing.T) {
+	ctx := context.Background()
+	rawLogger := log.NewFromOptions(logOptions)
+	logger := rawLogger.Sugar()
+
+	ctrlruntimelog.SetLogger(zapr.NewLogger(rawLogger.WithOptions(zap.AddCallerSkip(1))))
+
+	if err := credentials.Parse(); err != nil {
+		t.Fatalf("Failed to get credentials: %v", err)
+	}
+
+	seedClient, config, err := utils.GetClients()
+	if err != nil {
+		t.Fatalf("failed to get client for seed cluster: %v", err)
+	}
+
+	testJig := jig.NewBYOCluster(seedClient, logger, credentials)
+	testJig.ClusterJig.WithTestName("ear").WithFeatures(map[string]bool{
+		kubermaticv1.ClusterFeatureEtcdLauncher: true,
+	})
+
+	logger.Info("setting up the cluster")
+
+	_, cluster, err := testJig.Setup(ctx, jig.WaitForReadyPods)
+	defer testJig.Cleanup(ctx, t, true)
+	if err != nil {
+		t.Fatalf("failed to setup test environment: %v", err)
+	}
+
+	err = testJig.WaitForHealthyControlPlane(ctx, defaultTimeout)
+	if err != nil {
+		t.Fatalf("Cluster did not get healthy after enabling encryption-at-rest: %v", err)
+	}
+
+	userClient, err := testJig.ClusterClient(ctx)
+	if err != nil {
+		t.Fatalf("failed to create user cluster client: %v", err)
+	}
+
+	logger.Info("creating a dummy secret for testing encryption-at-rest")
+	secret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "dummy-",
+			Namespace:    "default",
+		},
+		Data: map[string][]byte{
+			"dummy-key": []byte("dummy-value"),
+		},
+	}
+
+	err = userClient.Create(ctx, &secret)
+	if err != nil {
+		t.Fatalf("failed to create secret: %v", err)
+	}
+
+	r := runner{
+		seedClient: seedClient,
+		userClient: userClient,
+		config:     config,
+		logger:     logger,
+		testJig:    testJig,
+	}
+
+	// Test Case 1:
+	// Enable encryption at rest and verify it works
+	logger.Info("Test Case 1 running...")
```

Comment:
A minor style note, given this is running end-to-end as what I would call a single test case (we aren't doing table testing or anything like that), I would suggest changing the log lines for this (and all the other "Test Case x running"). The comments above each lines are explaining well what is being tested, maybe we can put that information (in abbreviated form) into the logs?

Something like

```suggestion
	logger.Info("Enabling encryption at rest...")
```

And similar for later log lines.

Reply:
Thanks! removed redundant log statements in 6379af519a1da088688f2a62047439ad73d818c5. let me know if it needs further work